pkg/maps/ctmap/utils.go

```diff
@@ -172,7 +172,16 @@ func egressCTKeyFromEgressNatKey(k nat.NatKey) bpf.MapKey {
 	}
 }
 
-func ctEntryExist(ctMap *Map, ctKey bpf.MapKey) bool {
-	_, err := ctMap.Lookup(ctKey)
-	return !errors.Is(err, unix.ENOENT)
+func ctEntryExist(ctMap *Map, ctKey bpf.MapKey, f func(*CtEntry) bool) bool {
+	v, err := ctMap.Lookup(ctKey)
+
+	if err != nil {
+		return !errors.Is(err, unix.ENOENT)
+	}
```

Comment:
This looks like it would claim that a CT entry exists on all error cases except for explicit `ENOENT`.
A quick scan tells me other errors are unlikely in normal operation, and those errors would indicate a failed lookup. A nitpick I know, but wouldn't it be a safe bet to just return `false` on all errors?

Reply:
Yes this looked quirky. But I believe the idea was to err on the side of caution, and only remove NAT entries when we 100% know that the relevant CT entry no longer exists.